### ISSUE TITLE
Timeout (no response) when completing a large multipart upload

### ISSUE THREAD

**Issue by recunius** [MoriTanosuke/glacieruploader#64]
Hi,

I received a timeout on the completion of a 3.1TB file to Glacier using --multipartupload option.

I checked the treehash within my glacier VAULT and compared it to the treehash using `java -jar glacieruploader.jar -a FILE` and they match. So it appears the upload was successful and glacieruploader was just not notified of successful completion.

I have not encountered this issue with smaller uploads (I've tried 2TB and smaller in the past).

Is it possible to extend the timeout and/or update a keep alive setting on the connection in order to properly receive the response when finalizing a large upload?

Here are logs during the upload.

[multipart_complete_failed.log](https://github.com/MoriTanosuke/glacieruploader/files/1738038/multipart_complete_failed.log)

Thanks

**Comment by MoriTanosuke** [comment]
@recunius But the upload is complete and you can download a valid archive afterwards? If this is reproducible for larger archives, I might add a warning before the upload starts. There is no specific timeout and I don't see anything in the SDK documentation to set an additional keep-alive...

Hm, besides the error message _"The target server failed to respond"_ I can't see anything major in the logfile.

At https://docs.aws.amazon.com/amazonglacier/latest/dev/uploading-archive-mpu.html there is a table with some information about the glacier multipart uploads:

![auswahl_062](https://user-images.githubusercontent.com/105346/36410498-9928ff64-1611-11e8-8022-9d39a07f907e.png)

I don't think the archive size >3TB is causing an issue, but I can't be sure. Personally I never uploaded anything remotely near that archive size - and it's amazing to me that anyone is using my application like that. :smile:

**Comment by recunius** [comment]
@MoriTanosuke This software works really well. It's definitely the best maintained Glacier utility I've found that works within Linux.

For this archive I validated that the treehash listed in the AWS Glacier inventory matches the treehash that your tool calculates using the `-a` flag. I haven't downloaded the archive to be fully certain it matches. I suppose I should do that now, just to be sure that 1) it's the same and 2) that I indeed can download a file of that size.

I can imagine testing this tool on large files can be extremely difficult. Perhaps in the future I will split up large uploads into multiple files before sending to glacieruploader.
